app/components/form_components/autocomplete/sort.js

```diff
@@ -0,0 +1,48 @@
+// Returns an array of integers representing the position of each regex match.
+const matchPositions = (string, regexes) => regexes.map(regex => string.search(regex)).filter(i => i >= 0)
+
+const calculateWeight = (name, rawSynonyms, query) => {
```

Comment:
could make synonyms the last argument so we could default it to [] to avoid needing line 6

```suggestion
  const calculateWeight = (name, query, synonyms = []) => {
```